CMake/Autodownload.cmake

```diff
@@ -0,0 +1,84 @@
+function(get_deps LINK DEPS_NAME PACKAGE)
```

Comment:
Very nice!  Great to have this functionality split out.  It significantly simplifies the main `CMakeLists.txt` for sure.

A comment though; I know it is CMake and not C++, but would you mind adding some comments discussing what the function does?  We have quite a sizable CMake codebase at this point so we should do our best to keep it maintainable. :)